spec/reek/context/module_context_spec.rb

```diff
@@ -8,7 +8,7 @@
       module Fred
         def simple(x) x + 1; end
       end
-    ').to reek_of(:UncommunicativeParameterName, name: 'x')
+    ').to reek_of(:UncommunicativeParameterName, parameters: { name: 'x' })
```

Comment:
That sounds like a noble goal, but this is a reflection of the instance variables we have on smell_warning:

```ruby
      def initialize(smell_detector, context: '', lines: raise, message: raise,
                                     source: raise, parameters: {})
        @smell_detector = smell_detector
        @source         = source
        @context        = context.to_s
        @lines          = lines
        @message        = message
        @parameters     = parameters
      end
```

parameters is a hash - the others are not.  If we want to match against both the instance variables where one is a hash, we need to distinguish them with keys.  Maybe all the parameters should become instance variables to avoid the need for the nested parameters so to speak, but getting involved in this debate I feel a strong sense of "Yak-shaving" :-) I'm all for improving reek, but I really want to get to adding the internationalization functionality for which having the ability to check the message associated with a smell is a pre-requisite.

Maybe when making these changes is a great opportunity for a larger refactoring, but I don't feel qualified to do it.  Particularly I feel I am so far away from having an end user feature to drive the decisions that I can't really argue for or against this.